### PR TITLE
CAL-10 - Create thumbnails when ingesting a NITF

### PR DESCRIPTION
**Pull request by dcruver**
#### What does this PR do?
This PR adds thumbnails to ingested NITF products.

#### Who is reviewing it?
@kcwire 
@beyelerb 
@jlcsmith 

#### How should this be tested?
1. Install and start the alliance imaging application.
2. Log into DDF
3. Ingest a NITF
4. Do an empty search
    - verify that a thumbnail appears for the ingested NITF.

#### What are the relevant tickets?
CAL-10

#### Screenshots (if appropriate)
thumbnail-demo

#### Checklist:
- [x] Documentation Updated
- [x] Update / Add Unit Tests
- [x] Update / Add Integration Tests

Cal 10 - Create Nitf Input Transformer

### CODE REVIEW

File: catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java [hunk 1]
```diff
@@ -126,13 +134,20 @@ private void parseNitf(InputStream input, MetacardImpl metacard)
...
-                    (label) -> handleSegmentHeader(metacard, label,
-                            LabelAttribute.values()));
+                    (label) -> handleSegmentHeader(metacard, label, LabelAttribute.values()));
+
+            List<byte[]> imageSegmentData = parsingStrategy.getImageSegmentData();
+
+            if (imageSegmentData != null && imageSegmentData.size() > 0) {
+                NitfImageSegmentHeader thumbnailImageSegmentHeader = parsingStrategy
+                        .getImageSegmentHeaders().get(0);
+                byte[] thumbnailImageData = parsingStrategy.getImageSegmentData().get(0);
```
Comment (kcwire) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java]: Should we check the `byte[]` to make sure it's not empty?

File: catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java [hunk 2]
```diff
@@ -147,6 +162,33 @@ private void parseNitf(InputStream input, MetacardImpl metacard)
         }
     }
 
+    private void addThumbnail(Metacard metacard, NitfImageSegmentHeader header, byte[] image) {
+        try {
+            NitfRenderer renderer = new NitfRenderer();
```
Comment (kcwire) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java]: Can we reuse renderer? Is it thread safe?

File: catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java [hunk 3]
```diff
@@ -147,6 +162,33 @@ private void parseNitf(InputStream input, MetacardImpl metacard)
...
+            metacard.setAttribute(new AttributeImpl(Metacard.THUMBNAIL, thumbnailImage));
+        } catch (IOException|UnsupportedOperationException e) {
+            LOGGER.warn(e.getMessage(), e);
+        }
+    }
+
+    private byte[] scaleImage(final BufferedImage bufferedImage) throws IOException {
+        BufferedImage thumbnail = Thumbnails.of(bufferedImage)
+                .size(THUMBNAIL_WIDTH, THUMBNAIL_HEIGHT)
+                .outputFormat("jpg")
```
Comment (kcwire) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java]: nit but you should make `"jpg"` a constant.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java [hunk 4]
```diff
@@ -147,6 +162,33 @@ private void parseNitf(InputStream input, MetacardImpl metacard)
...
+            BufferedImage bufferedImage = renderer.render(header, inputStream);
+            byte[] thumbnailImage = scaleImage(bufferedImage);
+            metacard.setAttribute(new AttributeImpl(Metacard.THUMBNAIL, thumbnailImage));
+        } catch (IOException|UnsupportedOperationException e) {
+            LOGGER.warn(e.getMessage(), e);
+        }
+    }
+
+    private byte[] scaleImage(final BufferedImage bufferedImage) throws IOException {
+        BufferedImage thumbnail = Thumbnails.of(bufferedImage)
```
Comment (kcwire) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java]: Does Thumbnailtor throw any runtime exceptions we should protect against?

**Comment by kcwire** [comment]
Can you update the PR title to your ticket # and description please.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java (hunk 3 above)
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java]: done.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java (hunk 2 above)
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java]: It's threadsafe but small, so I think it's better to keep it in local-scope.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java (hunk 1 above)
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java]: done.

**Pull request merged**